ouroboros-consensus/src/Ouroboros/Storage/ChainDB/API.hs

```diff
@@ -348,6 +381,16 @@ data IteratorResult blk =
 deriving instance (Eq   blk, Eq   (HeaderHash blk)) => Eq   (IteratorResult blk)
 deriving instance (Show blk, Show (HeaderHash blk)) => Show (IteratorResult blk)
 
+-- | Deserialise the 'IteratorResult' of an iterator that yields serialised
+-- blocks.
+deserialiseIteratorResult
+  :: Monad m
+  => IteratorResult (Deserialisable m blk) -> m (IteratorResult blk)
+deserialiseIteratorResult = \case
+    IteratorExhausted      -> return $ IteratorExhausted
+    IteratorBlockGCed hash -> return $ IteratorBlockGCed hash
+    IteratorResult    blk  -> IteratorResult <$> deserialise blk
```

Comment:
Ah the `HeaderHash` prevents us from getting this for free with a `Functor` instance I suppose?

Reply:
Indeed :slightly_frowning_face: